src/event_sources/async_posix.cc

```diff
@@ -56,11 +56,14 @@ void AsyncEventThread::entry() {
     auto func = element->func;
     delete element;
     state_ = RUNNING;
+    word result;
```

Comment:
Maybe add a comment to explain why you want to be in the IDLE state before calling `on_event`?

Reply:
done.